build-image/Dockerfile

```diff
@@ -0,0 +1,63 @@
+FROM docker.io/bitnami/minideb:bullseye-amd64 as slimmed-rust
+
+# metadata
+LABEL io.parity.image.vendor="Parity Technologies" \
+    io.parity.image.title="contracts-verifiable" \
+    io.parity.image.documentation="https://github.com/paritytech/scripts/blob/master/\
+    dockerfiles/ink-ci-linux/README.md" \
+    io.parity.image.description="Inherits from docker.io/bitnami/minideb:bullseye. \
+    rust nightly, clippy, rustfmt, miri, rust-src, rustc-dev, grcov, rust-covfix, \
+    llvm-tools-preview, cargo-contract, xargo, binaryen, parallel, codecov, ink, solang" \
+    io.parity.image.documentation="https://github.com/paritytech/cargo-contract/blob/master/\
+    build-image/README.md"
+
+ENV RUSTUP_HOME=/usr/local/rustup \
+    CARGO_HOME=/usr/local/cargo \
+    PATH=/usr/local/cargo/bin:$PATH \
+    # The 1.69 toolchain is temporarily required to build ink! contracts because of
+    # https://github.com/paritytech/cargo-contract/issues/1139
+    RUST_VERSION=1.69
+
+# Minimal Rust dependencies.
+RUN set -eux \
+    && apt-get update && apt-get -y install wget \
+    && url="https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init" \
+    && wget "$url" \
+    && chmod +x rustup-init \
+    && ./rustup-init -y --no-modify-path --profile minimal --component rust-src rustfmt --default-toolchain $RUST_VERSION  \
+    && rm rustup-init \
+    && chmod -R a+w $RUSTUP_HOME $CARGO_HOME \
+    && rustup --version \
+    && cargo --version \
+    && rustc --version \
+    && apt-get remove -y --auto-remove wget \
+    && apt-get -y install gcc \
+    && rm -rf /var/lib/apt/lists/*
+
+FROM slimmed-rust as cc-builder
+
+# This is important, see https://github.com/rust-lang/docker-rust/issues/85
+ENV RUSTFLAGS="-C target-feature=-crt-static"
+
+RUN apt-get -y update && apt-get -y install gcc g++
+# Install cargo contract
+RUN cargo install cargo-contract
+
+# Cleanup after `cargo install`
+RUN rm -rf ${CARGO_HOME}/"registry" ${CARGO_HOME}/"git" /root/.cache/sccache
+
+# apt clean up
+RUN apt-get remove -y gnupg && \
+    apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
```

Comment:
FYI: Every command that writes unneeded files must be chained into a single `RUN` together with the cleanup commands, otherwise this won't work. Even if this is just a builder image it should be done correctly (those layers have to live somewhere on local storage too).

Reply:
Sure, for example [here](https://docs.docker.com/engine/reference/builder/#run)

Reply:
I've reread that page and can't find the confirmation for your words -- i.e. it _does_ mention ofc that this creates another layer and _might_ not work if previous layers don't install anything into these dirs but if they did, then this should work.

@SkymanOne , if you want to check for yourself there's a handy tool called `dive` https://github.com/wagoodman/dive where you can check how each layer modifies the container. If what @xermicus wrote is true, then there should be some stuff in `/var/lib/apt/lists/*` dir.

Reply:
Yes, indeed. `/var/lib/apt/lists` does not get removed. I will take @xermicus advice in 😄

Reply:
We have indeed reduced the intermediate layer size. I am going to publish the image now and we can push further releases if necessary